pinot-query-planner/src/main/java/org/apache/pinot/query/timeboundary/TimeBoundaryStrategy.java

```diff
@@ -43,4 +44,12 @@ public interface TimeBoundaryStrategy {
    */
   TimeBoundaryInfo computeTimeBoundary(LogicalTableConfig logicalTableConfig, TableCache tableCache,
       RoutingManager routingManager);
+
+
+  /**
+   * Returns the list of physical table names that are part of the time boundary.
+   * @param logicalTableConfig The logical table configuration
+   * @return The list of physical table names that are part of the time boundary.
+   */
+  List<String> getTimeBoundaryTableNames(LogicalTableConfig logicalTableConfig);
```

Comment:
This is specific to `MinTimeBoundaryStrategy`. Other strategies may not need it. Is a `cleanup(RouteEntryMap)` a better interface method ?

Reply:
My thinking, time boundary will be determined based on a set of [offline] tables though calculation could be min, max or something else.

I think, `_routingEntryMap` should not be passed to other classes and its management should remain with in `BrokerRoutingManager`. Also, each strategy need to implement the `cleanup` method and why it would be different for different strategy.

The time boundary specific tables will be required to build and remove routing, why it should be done in strategy?